### PR TITLE
Added a failure counter to the HTTP service discovery

### PR DESCRIPTION
**Pull request by David-N-Perkins**
For issue [10309](https://github.com/prometheus/prometheus/issues/10309).

### CODE REVIEW

**Comment by David-N-Perkins** [comment]
@roidelapluie Please review when you get a chance. Thanks!

**Comment by roidelapluie** [comment]
I would avoid setting the URL in labels, as it is not guaranteed to be unique.

I would contruct an identifier "jobname-ID"

**Comment by David-N-Perkins** [comment]
Not sure what you mean by "where ID is the name of the SD config used in the logs". Initially, I thought this was the value returned by `Config.Name()`, but I don't see how that would be a unique ID. Then I tried to find some other ID for a config used in log messages, but couldn't find anything. Could you be a little more specific about where to get the ID?

Also, should I still include the URL as a label or remove it all together? Just thinking it would be easier for users to identify based on the URL value.

**Comment by roidelapluie** [comment]
Indeed, it would be hard to get a unique identifier. I think we could keep it simple like consul, and not have a label. Consul has a simple failure counter. This is the safest, otherwise you could have duplicate metrics and therefore panics. Additionally, I am not sure we should expose such info's in /metrics endpoints.

**Comment by David-N-Perkins** [comment]
OK, I can do that. I'll post on the issue to make sure that meets the needs of the request.

File: discovery/http/http.go [hunk 1]
```diff
@@ -45,10 +46,17 @@ var (
 	}
 	userAgent        = fmt.Sprintf("Prometheus/%s", version.Version)
 	matchContentType = regexp.MustCompile(`^(?i:application\/json(;\s*charset=("utf-8"|utf-8))?)$`)
+
+	failuresCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "prometheus_sd_http_failures_total",
+			Help: "Number of refresh failures for an HTTP service discovery.",
```
Comment (roidelapluie) [inline discovery/http/http.go:53]:
```suggestion
			Help: "Number of HTTP service discovery refresh failures.",
```

**Review by LeviHarrison — APPROVED** [review]

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**